internal/terraform/node_resource_abstract_instance.go

```diff
@@ -926,13 +940,33 @@ func (n *NodeAbstractResourceInstance) plan(
 			ProposedNewState: proposedNewVal,
 			PriorPrivate:     priorPrivate,
 			ProviderMeta:     metaConfigVal,
+			DeferralAllowed:  ctx.Deferrals().DeferralAllowed(),
 		})
 	}
 	diags = diags.Append(resp.Diagnostics.InConfigBody(config.Config, n.Addr.String()))
 	if diags.HasErrors() {
 		return nil, nil, keyData, diags
 	}
 
+	if resp.Deferred != nil {
+		reqRep, reqRepDiags := getRequiredReplaces(priorVal, proposedNewVal, resp.RequiresReplace, n.ResolvedProvider.Provider, n.Addr)
+		diags = diags.Append(reqRepDiags)
+		if diags.HasErrors() {
+			return nil, nil, keyData, diags
+		}
+
+		action, _ := getAction(n.Addr, priorVal, resp.PlannedState, createBeforeDestroy, forceReplace, reqRep)
+		ctx.Deferrals().ReportResourceInstanceDeferred(n.Addr, resp.Deferred.Reason, &plans.ResourceInstanceChange{
```

Comment:
It's also the same here, we are missing lots of extra information in the change that is actually already available. Maybe we can just note that the change should be deferred, and then let the function compute the complete change like it does already and then log the deferral once at the end?